frappe/tests/test_db_query.py

```diff
@@ -166,6 +166,11 @@ def test_child_table_join(self):
 		self.assertIn("parent 1 child record 2", parent1_children)
 		self.assertEqual(results2[0].child_title, "parent 2 child record 1")
 
+	def test_allow_select_const(self):
+		hide_name = frappe.get_all("DocType", fields=["name", "'name' as `name`"])
```

Comment:
Any real use cases for this? 

Also `update` argument does exactly this.

Reply:
> Also `update` argument does exactly this.

Oh, haven't used this one before - thanks for the nudge. I'd prefer having the option to let the DBMS resolve this rather than iterating over a potentially huge list and mutating the records.

> Any real use cases for this?

I've reduced my use cases to simple examples in the description & test.

---

Any reason to not allow this in particular given how aliasing columns and functions are just fine? Don't see any implications for this change. Also, passing via update isn't as intuitive.

Reply:
No specific reason. If we wanna talk perf then this is just as worse if not more. 

While [doing this](https://github.com/frappe/frappe/pull/19810) I discovered that we double-buffer results because pymysql internally buffers results as tuples _separately_, we again do `as_dict` in most places. Also fetching copy of static string via network vs single static immutable string being referred by many rows. :woozy_face: String intering might kick in but idk if it makes any real difference here... so hardly any real benefit of doing it in query over `update`.

Reply:
No interning 

![image](https://user-images.githubusercontent.com/9079960/216616780-f174e833-2bc4-432c-9b12-5cbce1bdcb99.png)

_some_ reuse

![image](https://user-images.githubusercontent.com/9079960/216617150-3d7ec1de-7e50-4e72-bb99-9e7dd05a0b43.png)

Reply:
Wonder if I should add a warning instead or something as a nudge?

`warnings.warn("Constants should be passed as the update parameter instead of fields", stacklevel=3)`